src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -696,19 +698,28 @@ def _get_zstack_levels(self, focus_value):
         :returns (list(float) or None) zstack levels for zstack acquisition.
           return None if only one zstep is requested.
         """
-        if len(self._init_zlevels) == 1:
+        # When initial z level is one or None, use the current focus value
+        if len(self._init_zlevels) <= 1:
             return [focus_value, ]
 
         zlevels = self._init_zlevels + focus_value
 
-        # Clip zsteps value to allowed range
-        focus_range = self._focus_stream.focuser.getMetadata()[model.MD_POS_ACTIVE_RANGE]
-        axis_range = self._focus_stream.focuser.axes['z'].range
-        zmin = max(min(zlevels), min(axis_range), min(focus_range))
-        zmax = min(max(zlevels), max(axis_range), max(focus_range))
+        # Check which focus range is available
+        if self._focus_range is not None:
+            comp_range = self._focus_range
+        else:
+            comp_range = self._focus_stream.focuser.axes['z'].range
+
+        # The number of zlevels will remain the same, but the range will be adjusted
+        zmin = max(min(zlevels), min(comp_range))
+        zmax = min(max(zlevels), max(comp_range))
```

Comment:
That's not really useful. The distance between the zlevels should be fixed. Please use the same computation as in gui.win.acquistion._get_zstack_levels().
Essentially, it keeps the number of zlevels and the distance between zlevels. (by always adjusting both zmin and zmax simultaneously). Of course, this only works if (zmax-zmin) < (comp_range[1]-comp_range[0]), but that should always be the case.

Reply:
@tepals, to keep distance between the z stack levels same, I will have to use the code suggested above. Is there anything I should keep in mind related to above changes?

Reply:
there's nothing to worry about